### ISSUE TITLE
Whitelist Gitlab Raw Files

### ISSUE THREAD

**Issue by omtinez** [Microsoft/AzureNotebooks#46]
Just like Github raw files are whitelisted, can we whitelist Gitlab's equivalent? It was working yesterday, and today it suddenly stopped working.

As an aside, it would be nice to have a publicly available list of what's whitelisted. It's unfortunate that many valuable APIs are not and you seem unwilling to bulge on that (e.g. Youtube's API or Google APIs in general). To me, a more sensible approach like rate-limiting would make your tool far more valuable and actually replace my own Azure-hosted Jupyter server.

**Comment by crwilcox** [comment]
What is the specific URL you are speaking about?

**Comment by omtinez** [comment]
E.g.:

    > !curl https://gitlab.com/omtinez/combinatorics/raw/master/package.json -o package.json
    <   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current Dload  Upload   Total   Spent    Left  Speed
    <   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
    < curl: (56) Received HTTP code 403 from proxy after CONNECT

**Comment by crwilcox** [comment]
This is likely do-able. If we whitelist the domain gitlab.com is that enough to unblock you or are their subdomains that you need (api.gitlab.com, raw.gitlab.com, etc)?

Also, are there other sites you would need to complete your solution?

**Comment by omtinez** [comment]
Gitlab does not use a subdomain for raw files, so the top-level domain should do. As far as I can tell, their API is also in the top-level domain.

Other APIs / datastores nice to have whitelisted would be:
* https://api.500px.com
* https://api.flickr.com
* https://api.clarifai.com
* https://data.sfgov.org
* Any links found in https://catalog.data.gov/dataset

Some of those may be more prone to abuse than others, would you prefer if I opened separate issues for separate whitelist requests?

**Comment by DinoV** [comment]
Sorry this took so long but it should be working now.
